jqassistant/structure.adoc

```diff
@@ -0,0 +1,118 @@
+[[structure:Default]]
+[role=group,includesConstraints="structure:RootPackage,structure:FrontendDependency,structure:JobsDependency,structure:ServiceDependency"]
+== Structural Rules
+
+The package `net.sf.sze` is used as the root package of the application.
+
+[[structure:RootPackage]]
+.Labels the package `net.sf.sze` as `Root`.
+[source,cypher,role=concept]
+----
+MATCH
+  (root:Package)
+WHERE
+  root.fqn = "net.sf.sze"
+SET
+  root:Root
+RETURN
+  root as RootPackage
+----
+
+[[structure:RootPackage]]
+.All Java types must be located within the root package or a sub-package thereof.
+[source,cypher,role=constraint,requiresConcepts="structure:RootPackage,maven:InternalFile"]
+----
+MATCH
+  (type:Type:Internal)
+WHERE NOT
+  (:Package:Root)-[:CONTAINS*]->(type)
+RETURN
+  type as TypeOutsideRootPackage
+----
+
+All packages which are directly contained in the root package are technical slices.
+
+[[structure:Slice]]
+.Adds the label `Slice` to all packages contained in the root package (of all non-test artifacts).
+[source,cypher,role=concept,requiresConcepts="structure:RootPackage"]
+----
+MATCH
+  (artifact:Artifact)-[:CONTAINS]->(root:Package:Root)-[:CONTAINS]->(slice:Package)
+WHERE
+   artifact.type <> "test-jar"
```

Comment:
It's explained in the plugin docs of the Maven Project Scanner (http://buschmais.github.io/jqassistant/doc/1.2.0/#MavenProjectScanner), Section 8.11.1.2. "Nodes labled with :Maven:Artifact"